pkgs/applications/networking/p2p/qbittorrent/default.nix

```diff
@@ -1,44 +1,46 @@
-{ stdenv, fetchurl, pkgconfig, which
+{ stdenv, fetchFromGitHub, pkgconfig
 , boost, libtorrentRasterbar, qtbase, qttools, qtsvg
 , debugSupport ? false # Debugging
-, guiSupport ? true, dbus_libs ? null # GUI (disable to run headless)
+, guiSupport ? true, dbus ? null # GUI (disable to run headless)
 , webuiSupport ? true # WebUI
 }:
 
-assert guiSupport -> (dbus_libs != null);
-with stdenv.lib;
+assert guiSupport -> (dbus != null);
 
 stdenv.mkDerivation rec {
   name = "qbittorrent-${version}";
-  version = "4.1.0";
+  version = "4.1.1";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/qbittorrent/${name}.tar.xz";
-    sha256 = "0fdr74sc31x421sb69vlgal1hxpccjxxk8hrrzz9f5bg4jv895pw";
+  src = fetchFromGitHub {
+    owner = "qbittorrent";
+    repo = "qbittorrent";
+    rev = "release-${version}";
+    sha256 = "09bf1jr2sfdps8cb154gjw7zhdcpsamhnfbgacdmkfyd7qgcbykf";
   };
 
-  nativeBuildInputs = [ pkgconfig which ];
+  nativeBuildInputs = [ pkgconfig ];
 
-  buildInputs = [ boost libtorrentRasterbar qtbase qttools qtsvg ]
-    ++ optional guiSupport dbus_libs;
+  buildInputs = [ boost libtorrentRasterbar qtbase qttools qtsvg
+  ] ++ (if guiSupport then [ dbus ] else []); # D(esktop)-Bus depends on GUI support
```

Comment:
This evaluates correctly.

The comment should be the other way around, "GUI support depends on dbus", but `assert guiSupport -> (dbus != null);` above says exactly this, so the comment is redundant.

Please understand that the old version,
```
buildInputs = [ boost libtorrentRasterbar qtbase qttools qtsvg ]
  ++ optional guiSupport dbus;
```
is the most usual in Nixpkgs and it is more readable once you get accustomed. You may like the new version more now, but you will probably prefer the old version in the future, and in such cases as this essentially everyone writes `optional`. The way you've made `configureFlags` consistent is fine (although going in the other direction with `optionals (!guiSupport) [ "--disable-gui --enable-systemd" ] ++ optional (!webuiSupport) "--disable-webui"` is more conventional; this is a good example: https://github.com/NixOS/nixpkgs/blob/18.03/pkgs/applications/networking/mailreaders/claws-mail/default.nix#L69), but the old `buildInputs` are perfect and can not be improved.

Reply:
No. D-Bus features is additional option to the GUI build.
There are `--disable-qt-dbus` for Autotools and `DBUS={ON,OFF}` option for CMake, and that only for GUI build.

About `optional`s I agree.